source/per-stage-dns/lib/cicd-pipeline-stack.ts

```diff
@@ -0,0 +1,132 @@
+import {
+    Construct,
+    Stack,
+    StackProps,
+    Stage,
+    StageProps,
+    SecretValue,
+    CfnOutput
+} from "@aws-cdk/core";
+import { CdkPipeline, SimpleSynthAction } from "@aws-cdk/pipelines";
+import { Artifact } from "@aws-cdk/aws-codepipeline";
+import {
+    GitHubSourceAction,
+} from "@aws-cdk/aws-codepipeline-actions";
+import { config, SharedIniFileCredentials, Organizations } from "aws-sdk";
+import { PolicyStatement } from "@aws-cdk/aws-iam";
+import { DNSInfrastructureStack } from "./dns-infrastructure-stack";
+
+export class DNSInfrastructureStage extends Stage {
+    constructor(scope: Construct, id: string, props: StageProps) {
+        super(scope, id, props);
+        new DNSInfrastructureStack(this, "DNS-Infrastructure", {...props, stageName: id.toLowerCase()});
+    }
+}
+
+export class DNSInfrastructurePipelineStack extends Stack {
+    constructor(scope: Construct, id: string, props?: StackProps) {
+        super(scope, id, props);
+
+        const sourceArtifact = new Artifact();
+        const cloudAssemblyArtifact = new Artifact();
+        const oauth = SecretValue.secretsManager("GITHUB_TOKEN");
+
+        const pipeline = new CdkPipeline(this, "DNSInfrastructurePipeline", {
+            sourceAction: new GitHubSourceAction({
+                actionName: "GitHub",
+                output: sourceArtifact,
+                owner: this.node.tryGetContext("github_alias"),
+                repo: this.node.tryGetContext("github_repo_name"),
+                branch: this.node.tryGetContext("github_repo_branch"),
+                oauthToken: oauth,
+            }),
+            synthAction: SimpleSynthAction.standardNpmSynth({
+                sourceArtifact,
+                cloudAssemblyArtifact,
+                // If we are using the mono repo examples then adding the proper prefix
+                subdirectory: this.node.tryGetContext("github_repo_name") === 'aws-bootstrap-kit-examples' ? 'source/per-stage-dns' : '',
+                buildCommand: "npm run build",
+                rolePolicyStatements: [
+                    new PolicyStatement({
+                        actions: [
+                            "organizations:ListAccounts",
+                            "organizations:ListTagsForResource"
+                        ],
+                        resources: ["*"],
+                    }),
+                ],
+            }),
+            cloudAssemblyArtifact: cloudAssemblyArtifact,
+        });
+
+
+        new CfnOutput(this, "PipelineConsoleUrl", {
+            value: `https://${Stack.of(this).region}.console.aws.amazon.com/codesuite/codepipeline/pipelines/${pipeline.codePipeline.pipelineName}/view?region=${Stack.of(this).region}`,
+        });
+
+        const AWS_PROFILE = "cicd";
```

Comment:
Is it a good idea to hardcode a CLI profile name?

Reply:
Yeah that the way I can log to the account the appropriate permission ... this profile should having created through AWS Bootstrap Kit dev process